src/extensions/principalFieldDirectiveExtension.ts

```diff
@@ -7,6 +7,7 @@ export const principalFieldDirectiveExtension = (documentAST, result) => {
   let extensions = {}
   if (path.length && result.errors && result.errors.length) {
     const errors = result.errors.find(e => isEqual(e.path, path))
+    if (!errors) return extensions
```

Comment:
`errors` can be `undefined`, if there are _other_ errors occurring that aren't principal field ones. So we want to just return early. Added a red/green spec for this.